localstack-core/localstack/services/kms/provider.py

```diff
@@ -1334,6 +1339,29 @@ def list_resource_tags(
         kwargs = {"NextMarker": next_token, "Truncated": True} if next_token else {}
         return ListResourceTagsResponse(Tags=page, **kwargs)
 
+    @handler("RotateKeyOnDemand", expand=False)
+    def rotate_key_on_demand(
```

Comment:
it would be good to leave a TODO in the code indicating that we are not storing the previous key material, or in some way indicates that that feature is yet supported. 

Calling `RotateKeyOnDemand` in our implementation will prevent the user to decrypt any previously encoded material.